File: MaWPictures/src/main/java/us/mikeandwan/photos/prefs/CategoryDisplayPreference.java
package us.mikeandwan.photos.prefs;

import android.content.SharedPreferences;


public class CategoryDisplayPreference {
    private final SharedPreferences _sharedPrefs;


    public CategoryDisplayPreference(SharedPreferences sharedPrefs) {
        _sharedPrefs = sharedPrefs;
    }


    public CategoryDisplay getCategoryDisplay() {
        String val = _sharedPrefs.getString("category_view_mode", "list");

        if(val.equalsIgnoreCase("list")) {
            return CategoryDisplay.ThumbnailAndNameList;
        }

        return CategoryDisplay.ThumbnailGrid;
    }
}

File: MaWPhotos/src/main/java/us/mikeandwan/photos/prefs/CategoryDisplayPreference.kt
package us.mikeandwan.photos.prefs

import android.content.SharedPreferences

class CategoryDisplayPreference(private val _sharedPrefs: SharedPreferences) {
    val categoryDisplay: CategoryDisplay
        get() {
            val value = _sharedPrefs.getString("category_view_mode", "list")
            return if (value == null || value.equals("list", ignoreCase = true)) {
                CategoryDisplay.List
            } else CategoryDisplay.Grid
        }
}